### ISSUE TITLE
Axis type updates to logarithmic and back to linear is buggy

### ISSUE THREAD

**Issue by govinda18** [highcharts/highcharts#15461]
#### Expected behaviour
Axis type updates should work fine

#### Actual behaviour
Updating axis type from linear to logarithmic and back to linear seems to be buggy

#### Live demo with steps to reproduce
https://jsfiddle.net/kjhdt1pe/1/

#### Product version
Highcharts JS v9.0.1 (2021-02-15)

#### Affected browser(s)
Browser

**Comment by KacperMadej** [comment]
Hi @govinda18 

This is by design as described in [the docs](https://www.highcharts.com/docs/chart-concepts/axes#logarithmic): "Highcharts will remove 0 or negative points associated to the axis".

Using mixed positive and negative data on a logarithmic axis requires additional support for [zero in a logarithmic axis](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/yaxis/type-log-negative/). More info in [this API reference](https://api.highcharts.com/highcharts/xAxis.type).

tl;dr here it works - https://jsfiddle.net/BlackLabel/w1L6exad/
